mongodbatlas/data_source_mongodbatlas_search_index_test.go

```diff
@@ -0,0 +1,43 @@
+package mongodbatlas
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"os"
+	"testing"
+)
+
+func TestAccDataSourceMongoDBAtlasSearchIndex_byID(t *testing.T) {
+	var (
+		clusterName = acctest.RandomWithPrefix("test-acc-global")
+		projectID   = os.Getenv("MONGODB_ATLAS_PROJECT_ID")
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckMongoDBAtlasSearchIndexDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccMongoDBAtlasSearchIndexDSConfig(projectID, clusterName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet("mongodbatlas_search_index.test", "name"),
+					resource.TestCheckResourceAttrSet("mongodbatlas_search_index.test", "project_id"),
+					resource.TestCheckResourceAttrSet("mongodbatlas_search_index.test", "cluster_name"),
+				),
+			},
+		},
+	})
+}
+
+func testAccMongoDBAtlasSearchIndexDSConfig(projectID string, clusterName string) string {
+	return fmt.Sprintf(`
+		%s
+
+		data "mongodbatlas_search_index" "test" {
```

Comment:
can we set some indexes here and check if they are set?

Reply:
sure!